### PR TITLE
Np to vtk cnvrt

### PR DESCRIPTION
**Pull request by giltis**
Added tools for converting numpy arrays to vtk Objects, as well as the opposite (vtk Objects to numpy arrays).

Additional tools and examples to follow.

### CODE REVIEW

**Comment by tacaswell** [comment]
Aren't a large number of these files already in other PRs?

File: skxray/api/image_processing/morphology.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+# Copyright (c) 2014, Brookhaven Science Associates, Brookhaven        #
+# National Laboratory. All rights reserved.                            #
+#                                                                      #
+# @author: Li Li (lili@bnl.gov)                                        #
+# created on 07/10/2014                                                #
+#                                                                      #
+# Original code:                                                       #
+# @author: Mirna Lerotic, 2nd Look Consulting                          #
+#         http://www.2ndlookconsulting.com/                            #
+# Copyright (c) 2013, Stefan Vogt, Argonne National Laboratory         #
```
Comment (ericdill) [inline skxray/api/image_processing/morphology.py]: I'm assuming that this block of the copyright is wrong... Can you remove this?

**Comment by ericdill** [comment]
I'm going to make a few restructuring suggestions, but after those have been done I would advocate for simply merging this PR and dealing with issues as they crop up... @tacaswell I know you're not going to like this very much, but this PR is literally never going to get merged otherwise

**Comment by ericdill** [comment]
My vision of what would go in the `skxray/api/` folder was science namespaces. With that in mind, I'd like to collapse **all** of the `skxray/api/image_processing/*` files into a single `skxray/api/image_processing.py` module

**Comment by ericdill** [comment]
I would like to see the `skxray/img_proc/` folder renamed to `skxray/image/`

**Comment by ericdill** [comment]
Rename `skxray/img_proc/imgQuant.py` to `skxray/img_proc/quant.py`

File: skxray/io/RAW_io.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
+# Module for the BNL image processing project
+# Developed at the NSLS-II, Brookhaven National Laboratory
+# Developed by Gabriel Iltis, Nov. 2013
+"""
+RAW (.raw and .volume)
+"""
+
+import numpy as np
+
+def read_RAW(file_name,
```
Comment (ericdill) [inline skxray/io/RAW_io.py]: RAW -> raw

File: skxray/io/RAW_io.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# Module for the BNL image processing project
+# Developed at the NSLS-II, Brookhaven National Laboratory
+# Developed by Gabriel Iltis, Nov. 2013
+"""
+RAW (.raw and .volume)
```
Comment (ericdill) [inline skxray/io/RAW_io.py]: Where do ".raw" and ".volume" files come from?

Comment (ericdill) [inline skxray/io/RAW_io.py]: nevermind, we will deal with this after this horrible monster gets merged

File: skxray/io/RAW_io.py (hunk 2 above)
Comment (ericdill) [inline skxray/io/RAW_io.py]: also nevermind on this one

**Comment by ericdill** [comment]
Rename `skxray/io/RAW_io.py` to skxray/io/raw.py

**Comment by ericdill** [comment]
Many of your files in the `io` directory have `_io` appended to them. Can you get rid of the `_io` appendage from these? It is unneeded as they are already in the `io` directory

**Comment by ericdill** [comment]
Can you rename `skxray/io/vtk_tools` to `skxray/io/vtk`?

**Comment by tacaswell** [comment]
Is there anyway to chop this up into a number of smaller PRs?  2k lines is an absolute bear to review.

**Comment by giltis** [comment]
@tacaswell : Holy smokes. I'm suspect that all of those additional commits were included because I probably created the branch while in another branch. I'll fix that because I meant for this PR to ONLY focus on numpy/VTK data conversion, with the next PR(s) to focus on VTK visualization and quantification tools.

**Comment by giltis** [comment]
The current commit generates a segmentation fault when an additional operation is executed on the "converted" data/array. This suggests that there is a problem with the conversion process and that the conversion is NOT currently working as it is supposed to.
Currently writing a conversion test that evaluates whether a given data set, and data type converts correctly.

File: skxray/io/api.py [hunk 4]
```diff
@@ -39,17 +39,21 @@
 import logging
 logger = logging.getLogger(__name__)
 
-from .net_cdf_io import load_netCDF
+from skxray.io.net_cdf_io import read_netCDF
```
Comment (tacaswell) [inline skxray/io/api.py]: why did you de-relative import these?

**Comment by giltis** [comment]
Need to get in touch with kitware. Contact them about errors associated with numpy_support.np_to_vtk

**Comment by danielballan** [comment]
Take a look at the docstring as well. (It mentions the segfaults.)

**Comment by tacaswell** [comment]
@giltis Any progress on this?

**Comment by danielballan** [comment]
@giltis Any luck avoiding the segfaults by following the docstrings? Once we know it has worked for someone in our group once, we can check this off our list and thank VTK for doing it for us.

**Comment by licode** [comment]
maybe this one can be closed?

**Pull request closed without merging**